mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -340,6 +341,27 @@ export const WayPage = (props: WayPageProps) => {
         onChangeFinish={(text) => changeWayName(way, text)}
         isEditable={isOwner}
       />
+      {isOwner &&
+        <Button
+          value="Delete way"
+          onClick={() => {
+
+            /**
+             * CallBack triggered on press ok
+             */
+            const onOk = async () => {
+              await WayDAL.deleteWay(way);
+              navigate(pages.user.getPath({uuid: way.owner.uuid}));
+            };
+
+            renderModalContent({
+              description: way.name,
+              text: "way",
+              onOk,
+            });
+          }}
```

Comment:
Maybe it would be better to take function out?

Reply:
We will do refactor after use modal instead of confirm

Reply:
If we will make refactoring in future it does not mean we should not push clean code to main. Looks like this refactoring could be implemented before modal refactoring.

Reply:
Which function do you want to move out? onOk?

Reply:
Mobed out